conda/cli/main.py

```diff
@@ -147,6 +147,41 @@ def main(*args, **kwargs):
             return ExceptionHandler().handle_exception(exc_val, exc_tb)
 
     from ..exceptions import conda_exception_handler
+
+    # Here we are detecting (badly) the case when conda.bat has been run but not
+    # to activate or deactivate. conda.bat will have added entries for sys.prefix
+    # to the front of PATH (so that this conda executable can find its DLLs), but
+    # that is problematic for conda run, if executed through conda.bat because
+    # we do not prefix PATH with the new prefixes entries; we swap them in-place
+    # (a-la reactivate) or may even do nothing at all when re-activating the
+    # currently active env.
+    #
+    # I do not want to be messing with PATH here at all though (at least not until
+    # conda has loaded every DLL it could possibly want). We need a new proxy for
+    # os.environ['PATH'] which gets used for any PATH qeuries made by conda and
+    # which strips sysp from the result (which would be emitted to all script files
+    # (that need to set the PATH env var) and all subprocess envs.
+    #
+
+    from os import environ, pathsep, sep
+    if 'CONDA_PREFIX' in environ:
```

Comment:
Naive question since I'm just jumping back in... Should this be gated for `on_win` only?

Reply:
It wouldn't really matter, arguably we could be more uniform and force the sys prefix paths to the front of PATH in the unix scripts too and remove it on all if discovered there. Still, I'd like a better way to know if we've been called from the the wrapper script or now. I'll think about ways. Suggestions welcome!